libraries/AP_Scripting/examples/crosstrack_restore.lua

```diff
@@ -0,0 +1,47 @@
+--[[
+
```

Comment:
Fixed

Reply:
Sorry, I should have been more clear. 

I meant more that the script should not be used except on plane. It will run fine on other vehicles. As your comment points warns the functionality won't work. But it will change modes just fine. Except it won't be loiter and auto on other vehicles. It will be whatever modes happen have the same number.

You could check in the script and error out if not plane. I think this should work:
```
assert(FWVersion:type() == 3, "Must be plane")
```

Reply:
This will throw and error but then the scheduler will try to run the script again. I used an if statement instead which only runs update() if the test passes.